app/enc/uri/url.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 The Drivers Cooperative. All rights reserved.
+// Use of this source code is governed by a dual
+// license that can be found in the LICENSE file.
+
+package uri
```

Comment:
imo, this is very cool uri normalisation code that deserves a placement in proj root. like openmarketplaceengine/uri.

Reply:
nit: My preference would be to stick as closely as possible to this (de facto) [standard](https://github.com/golang-standards/project-layout). So every bit of Go code (with the exception of `main.go`) would probably end up in `cmd`, `internal`, or `pkg`.

**If** we do this, we should do so in a later PR since it would be a very large refactor.